src/expr-derive-impl/src/sqlfunc.rs

```diff
@@ -0,0 +1,441 @@
+        negate,
+        could_error,
+        propagates_nulls,
+    } = modifiers;
+
+    if is_infix_op.is_some() {
+        return Err(darling::Error::unknown_field("is_infix_op"));
+    }
+    if negate.is_some() {
+        return Err(darling::Error::unknown_field("negate"));
+    }
+    if propagates_nulls.is_some() {
+        return Err(darling::Error::unknown_field("propagates_nulls"));
+    }
+
+    let preserves_uniqueness_fn = preserves_uniqueness.map(|preserves_uniqueness| {
+        quote! {
+            fn preserves_uniqueness(&self) -> bool {
+                #preserves_uniqueness
+            }
+        }
+    });
+
+    let inverse_fn = inverse.as_ref().map(|inverse| {
+        quote! {
+            fn inverse(&self) -> Option<crate::UnaryFunc> {
+                #inverse
+            }
+        }
+    });
+
+    let is_monotone_fn = is_monotone.map(|is_monotone| {
+        quote! {
+            fn is_monotone(&self) -> bool {
+                #is_monotone
+            }
+        }
+    });
+
+    let name = if let Some(sqlname) = sqlname {
+        quote! {
+            #sqlname
+        }
+    } else {
+        quote! {
+            stringify!(#fn_name)
+        }
+    };
+
+    let (output_type, introduces_nulls_fn) = if let Some(output_type) = output_type {
+        let introduces_nulls_fn = quote! {
+            fn introduces_nulls(&self) -> bool {
+                <#output_type as mz_repr::DatumType<'_, ()>>::nullable()
+            }
+        };
+        let output_type = quote! { <#output_type> };
+        (output_type, Some(introduces_nulls_fn))
+    } else {
+        (quote! { Self::Output }, None)
+    };
+
+    let arena = arena.then(|| quote! { , temp_storage });
+
+    let could_error_fn = could_error.map(|could_error| {
+        quote! {
+            fn could_error(&self) -> bool {
+                #could_error
+            }
+        }
+    });
+
+    let result = quote! {
+        #[derive(proptest_derive::Arbitrary, Ord, PartialOrd, Clone, Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize, Hash, mz_lowertest::MzReflect)]
+        pub struct #struct_name;
+
+        impl<'a> crate::func::EagerUnaryFunc<'a> for #struct_name {
+            type Input = #input_ty;
+            type Output = #output_ty;
+
+            fn call(&self, a: Self::Input) -> Self::Output {
+                #fn_name(a #arena)
+            }
+
+            fn output_type(&self, input_type: mz_repr::ColumnType) -> mz_repr::ColumnType {
+                use mz_repr::AsColumnType;
+                let output = #output_type::as_column_type();
+                let propagates_nulls = crate::func::EagerUnaryFunc::propagates_nulls(self);
+                let nullable = output.nullable;
+                // The output is nullable if it is nullable by itself or the input is nullable
+                // and this function propagates nulls
+                output.nullable(nullable || (propagates_nulls && input_type.nullable))
```

Comment:
What about `introduces_nulls`?

Reply:
I _think_ it's handled via the output type, but I'm honestly not sure. I think I'll merge as-is, and revisit next.

Reply:
I don't know if you would ever want to expand this to variadic functions but it might be nice to take `introduces_nulls` as a parameter because the output type of `string_to_array` is not nullable but it can return nulls iff the first parameter is null

Reply:
Makes sense! I added a `introduces_nulls` parameters, which likely needs refinement in a later PR, but we should def. have it.